<a href="https://colab.research.google.com/github/VeraaaaWei/data_analytics/blob/Python/Lai_Unit3_Python_Advanced_Pandas_DataWrangling_Assignment.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Assignment 1 : Python Data Analysis using salaries dataset

# Data Exploration:  This should include summary statistics, means, medians, quartiles, or any other relevant information about the data set. 

In [51]:
import pandas as pd 
import numpy as np
import matplotlib.pyplot as plt
%matplotlib inline

In [67]:
df = pd.read_csv('https://raw.githubusercontent.com/CunyLaguardiaDataAnalytics/datasets/master/Salaries.csv')

**1. Analyze the dataset**
1.Provide your observation and conclusions




In [ ]:
type(df)

pandas.core.frame.DataFrame

In [ ]:
df.shape

(397, 7)

In [19]:
df.head()

Unnamed: 0      rank discipline  yrs.since.phd  yrs.service   sex  salary
0           1      Prof          B             19           18  Male  139750
1           2      Prof          B             20           16  Male  173200
2           3  AsstProf          B              4            3  Male   79750
3           4      Prof          B             45           39  Male  115000
4           5      Prof          B             40           41  Male  141500

In [ ]:
df.tail()

Unnamed: 0      rank discipline  yrs.since.phd  yrs.service   sex  salary
392         393      Prof          A             33           30  Male  103106
393         394      Prof          A             31           19  Male  150564
394         395      Prof          A             42           25  Male  101738
395         396      Prof          A             25           15  Male   95329
396         397  AsstProf          A              8            4  Male   81035

In [ ]:
df.describe()

Unnamed: 0  yrs.since.phd  yrs.service         salary
count  397.000000     397.000000   397.000000     397.000000
mean   199.000000      22.314861    17.614610  113706.458438
std    114.748275      12.887003    13.006024   30289.038695
min      1.000000       1.000000     0.000000   57800.000000
25%    100.000000      12.000000     7.000000   91000.000000
50%    199.000000      21.000000    16.000000  107300.000000
75%    298.000000      32.000000    27.000000  134185.000000
max    397.000000      56.000000    60.000000  231545.000000

# Mean/median of all salaries


1.   Find the Mean and Median of all Salaries
2.  Provide your findings



In [38]:
# Mean of all salaries

df.mean()

Unnamed: 0          199.000000
yrs.since.phd        22.314861
yrs.service          17.614610
salary           113706.458438
dtype: float64

In [39]:
# Median of all salaries
df.median()

Unnamed: 0          199.0
yrs.since.phd        21.0
yrs.service          16.0
salary           107300.0
dtype: float64

In [ ]:
#Findings: Instead of the codes above, we can also know the median and mean from the #df.describe(). 
#Mean is larger than Median here, which means it's left-skew. 

# Median salary between gender and/or rank(title) Etc...


Making a new Data Frame that only has the columns 'sex' and 'salary'

In [ ]:
df.columns
column_to_be_dropped = ['Unnamed: 0', 'rank', 'discipline', 'yrs.since.phd', 'yrs.service']
new_df = df.drop(column_to_be_dropped, axis = 1)
new_df.tail()


sex  salary
392  Male  103106
393  Male  150564
394  Male  101738
395  Male   95329
396  Male   81035

Finding the Median salary for all males

In [ ]:
new_df['sex'].is_unique


False

In [ ]:
new_df3 = new_df[ new_df['sex'] == 'Male']
new_df3.mean()

salary    115090.418994
dtype: float64

Finding the Median salary for all females

In [ ]:
new_df2 = new_df[ new_df['sex'] == 'Female']
new_df2.mean()

salary    101002.410256
dtype: float64

Comparing the Mean salary of Males and Females

In [ ]:
a = new_df3.mean() - new_df2.mean()
if int(a)>0:
  print('Males have high Mean salary, which is ' + str(int(a)) + ' higher than females.')
else:
  print('Females have high Mean salary, which is ' + str(int(a)) + ' higher than males.')

Males have high Mean salary, which is 14088 higher than females.


In [ ]:
# The difference between mean salaries of male and female

# Males have high Mean salary than females.


# Data Wrangling – perform some basic transformations on the data

People who have 20 years of experience (Total: 157), (Mean Salary: 122,520.66)

In [5]:
new_df20y = df[ df['yrs.service'] >= 20]
new_df20y

Unnamed: 0  rank discipline  yrs.since.phd  yrs.service   sex  salary
3             4  Prof          B             45           39  Male  115000
4             5  Prof          B             40           41  Male  141500
6             7  Prof          B             30           23  Male  175000
7             8  Prof          B             45           45  Male  147765
8             9  Prof          B             21           20  Male  119250
..          ...   ...        ...            ...          ...   ...     ...
384         385  Prof          A             27           21  Male  125192
386         387  Prof          A             29           27  Male  139219
388         389  Prof          A             38           36  Male  119450
392         393  Prof          A             33           30  Male  103106
394         395  Prof          A             42           25  Male  101738

[157 rows x 7 columns]

In [6]:
type(new_df20y)

pandas.core.frame.DataFrame

In [7]:
new_df20y.head()

Unnamed: 0  rank discipline  yrs.since.phd  yrs.service   sex  salary
3           4  Prof          B             45           39  Male  115000
4           5  Prof          B             40           41  Male  141500
6           7  Prof          B             30           23  Male  175000
7           8  Prof          B             45           45  Male  147765
8           9  Prof          B             21           20  Male  119250

In [8]:
new_df20y.tail()

Unnamed: 0  rank discipline  yrs.since.phd  yrs.service   sex  salary
384         385  Prof          A             27           21  Male  125192
386         387  Prof          A             29           27  Male  139219
388         389  Prof          A             38           36  Male  119450
392         393  Prof          A             33           30  Male  103106
394         395  Prof          A             42           25  Male  101738

In [9]:
new_df20y.describe()

Unnamed: 0  yrs.since.phd  yrs.service         salary
count  157.000000     157.000000   157.000000     157.000000
mean   213.286624      34.242038    31.057325  122520.656051
std    116.841173       8.392152     8.633746   29674.689571
min      4.000000      19.000000    20.000000   57800.000000
25%    114.000000      28.000000    23.000000  101036.000000
50%    232.000000      34.000000    30.000000  119450.000000
75%    308.000000      40.000000    37.000000  142500.000000
max    395.000000      56.000000    60.000000  231545.000000

Comparing the salaries of people who have 20+ years of experience vs less than 20 years of experience

In [10]:
new_df20y.columns
column_to_be_dropped1 = ['Unnamed: 0', 'rank', 'discipline', 'yrs.since.phd', 'sex']
new_df20y1 = df.drop(column_to_be_dropped1, axis = 1)
new_df20y1

yrs.service  salary
0             18  139750
1             16  173200
2              3   79750
3             39  115000
4             41  141500
..           ...     ...
392           30  103106
393           19  150564
394           25  101738
395           15   95329
396            4   81035

[397 rows x 2 columns]

In [11]:
new_df20y1plus = new_df20y1[ new_df20y1['yrs.service'] > 20]
column_to_be_dropped2 = ['yrs.service']
new_df20y11 = new_df20y1plus.drop(column_to_be_dropped2, axis = 1)
new_df20y11

new_df20y1minus = new_df20y1[ new_df20y1['yrs.service'] < 20]
column_to_be_dropped3 = ['yrs.service']
new_df20y22 = new_df20y1minus.drop(column_to_be_dropped2, axis = 1)
new_df20y22

new_df20y11.mean()


salary    122103.877551
dtype: float64

In [ ]:
new_df20y22.mean()


salary    107940.504167
dtype: float64

In [ ]:
b = new_df20y11.mean() - new_df20y22.mean()
if int(b)>0:
  print('People who have 20+ years of experiencehave high Mean salary, which is ' + str(int(b)) + ' higher than people less than 20 years of experience.')
else:
  print('People who have 20- years of experiencehave high Mean salary, which is ' + str(int(b)) + ' higher than people more than 20 years of experience.')

People who have 20+ years of experiencehave high Mean salary, which is 14163 higher than people less than 20 years of experience.


People who have salary of greater than or equal to 200,000

In [3]:
new_df200k = df[ df['salary'] >= 200000]
new_df200k

Unnamed: 0  rank discipline  yrs.since.phd  yrs.service   sex  salary
43           44  Prof          B             38           38  Male  231545
249         250  Prof          A             29            7  Male  204000
364         365  Prof          A             43           43  Male  205500

Your Findings here...
Concerning the mean salary, mostly the longer the years of experience you have, the higher your salary is. But in some individual situation, sometimes people with less years of experience has higher salary than the one with more experience, like the row 249 prof.


# Rename values, subset columns, etc

In [68]:
df.replace({'AsstProf':'Assistant prof'}, inplace= True)

df.head()

Unnamed: 0            rank discipline  ...  yrs.service   sex  salary
0           1            Prof          B  ...           18  Male  139750
1           2            Prof          B  ...           16  Male  173200
2           3  Assistant prof          B  ...            3  Male   79750
3           4            Prof          B  ...           39  Male  115000
4           5            Prof          B  ...           41  Male  141500

[5 rows x 7 columns]

In [69]:
df.rename(columns = {"discipline": "Disciplines"}, inplace= True)
df.head()

Unnamed: 0            rank Disciplines  ...  yrs.service   sex  salary
0           1            Prof           B  ...           18  Male  139750
1           2            Prof           B  ...           16  Male  173200
2           3  Assistant prof           B  ...            3  Male   79750
3           4            Prof           B  ...           39  Male  115000
4           5            Prof           B  ...           41  Male  141500

[5 rows x 7 columns]

# Create some simple visualizations showing some simple statistics

In [71]:
disb_df = df[df['Disciplines'] == 'B']
disb_df

Unnamed: 0            rank Disciplines  ...  yrs.service   sex  salary
0             1            Prof           B  ...           18  Male  139750
1             2            Prof           B  ...           16  Male  173200
2             3  Assistant prof           B  ...            3  Male   79750
3             4            Prof           B  ...           39  Male  115000
4             5            Prof           B  ...           41  Male  141500
..          ...             ...         ...  ...          ...   ...     ...
351         352            Prof           B  ...           38  Male   93519
352         353            Prof           B  ...           27  Male  142500
353         354            Prof           B  ...           20  Male  138000
354         355  Assistant prof           B  ...            1  Male   83600
355         356            Prof           B  ...           21  Male  145028

[216 rows x 7 columns]

In [72]:
disa_df = df[df['Disciplines'] == 'A']
disa_df

Unnamed: 0            rank Disciplines  ...  yrs.service     sex  salary
17           18            Prof           A  ...           34    Male  103450
18           19            Prof           A  ...           23    Male  124750
19           20            Prof           A  ...           36  Female  137000
20           21            Prof           A  ...           26    Male   89565
21           22            Prof           A  ...           31    Male  102580
..          ...             ...         ...  ...          ...     ...     ...
392         393            Prof           A  ...           30    Male  103106
393         394            Prof           A  ...           19    Male  150564
394         395            Prof           A  ...           25    Male  101738
395         396            Prof           A  ...           15    Male   95329
396         397  Assistant prof           A  ...            4    Male   81035

[181 rows x 7 columns]

In [73]:
disa_df.describe()

Unnamed: 0  yrs.since.phd  yrs.service         salary
count  181.000000     181.000000   181.000000     181.000000
mean   249.104972      25.381215    19.950276  108548.430939
std    102.951526      13.117989    13.678156   30538.145647
min     18.000000       2.000000     0.000000   57800.000000
25%    140.000000      14.000000     8.000000   83000.000000
50%    266.000000      27.000000    19.000000  104350.000000
75%    311.000000      36.000000    30.000000  125192.000000
max    397.000000      56.000000    57.000000  205500.000000

In [75]:
disb_df.describe()

Unnamed: 0  yrs.since.phd  yrs.service         salary
count  216.000000     216.000000   216.000000     216.000000
mean   157.013889      19.745370    15.657407  118028.694444
std    107.209878      12.135475    12.103170   29459.135212
min      1.000000       1.000000     0.000000   67559.000000
25%     64.750000      10.000000     5.750000   94905.250000
50%    157.500000      18.500000    14.000000  113018.500000
75%    211.250000      28.000000    23.000000  139836.500000
max    356.000000      56.000000    60.000000  231545.000000

Your observations from the visuals: 
The professors who teach discipline B have higher mean salary than the ones who teach discipline A.

# Draw some conclusions based on the data.  What did you discover?

Based on the analysis performed above, the conclusions are as follows:

Gender equality still has a long way to go.



I also discovered that the experience can become a kind of human resource to make more wealth.





